### PR TITLE
rg-129: Change Profile Data

### PR DESCRIPTION
**Pull request by Valentyn13**

### CODE REVIEW

**Review by MaxPoplavskiy — APPROVED** [review]

**Review by honkster-nazarii — APPROVED** [review]

File: frontend/src/bundles/profile/store/actions.ts [hunk 1]
```diff
@@ -1,24 +1,27 @@
 import { createAsyncThunk } from '@reduxjs/toolkit';
-import { type SocialMediaProfiles } from 'shared/build';
+import {
+    type Profile,
+    type SocialMediaProfiles,
+    type UpdateUserProfileAndEmailRequestDto,
+} from 'shared/build';
 
+import { actions } from '~/bundles/auth/store/slice.js';
```
Comment (vlad-blip) [inline frontend/src/bundles/profile/store/actions.ts:8]:
```suggestion
import { actions as authActions } from '~/bundles/auth/store/slice.js';
```
Maybe say explicitly that these are authActions.

File: frontend/src/bundles/profile/store/slice.ts [hunk 2]
```diff
@@ -33,9 +33,19 @@ const { reducer, actions, name } = createSlice({
             state.profile = action.payload;
             state.dataStatus = DataStatus.FULFILLED;
         });
-        builder.addMatcher(isAnyOf(disconnectSocialMedia.pending), (state) => {
-            state.dataStatus = DataStatus.PENDING;
+        builder.addCase(updateProfileAndEmail.fulfilled, (state, action) => {
```
Comment (vlad-blip) [inline frontend/src/bundles/profile/store/slice.ts:36]: It looks like copy of code above. Consider using `addMatcher` and `isAnyOf`

File: backend/src/bundles/users/user.service.ts [hunk 3]
```diff
@@ -105,6 +106,24 @@ class UserService
...
+    ): Promise<UserWithProfileRelation> {
+        const { profileId } = await this.userRepository.updateById(id, {
+            email,
+        });
+        await this.profileRepository.updateById(profileId, {
+            firstName,
+            lastName,
+        });
+
+        return this.userRepository.getUserWithProfileAndOauthConnections(
```
Comment (Icerock10) [inline backend/src/bundles/users/user.service.ts:121]:
```suggestion
   return this.getUserWithProfileAndOauthConnections(id);
```

File: frontend/src/bundles/profile/store/actions.ts (hunk 1 above)
Comment (Valentyn13) [reply frontend/src/bundles/profile/store/actions.ts]: done

File: frontend/src/bundles/profile/store/slice.ts (hunk 2 above)
Comment (Valentyn13) [reply frontend/src/bundles/profile/store/slice.ts]: done

File: frontend/src/bundles/users/pages/profile.tsx [hunk 4]
```diff
@@ -33,15 +32,23 @@ const Profile: React.FC = () => {
...
             firstName: auth.user?.userProfile.firstName as string,
             lastName: auth.user?.userProfile.lastName ?? '',
             email: auth.user?.email as string,
+            id: auth.user?.id as string,
         }),
     );
 
+    const handleFormSubmit = useCallback(
+        (payload: UpdateUserProfileAndEmailRequestDto) => {
+            void dispatch(updateProfileAndEmail({ id, payload }));
```
Comment (Icerock10) [inline frontend/src/bundles/users/pages/profile.tsx:47]:
```suggestion
            void dispatch(profileActions.updateProfileAndEmail({ id, payload }));
```

File: backend/src/bundles/users/user.controller.ts [hunk 5]
```diff
@@ -141,6 +160,23 @@ class UserController extends Controller {
...
+            params: {
+                id: string;
+            };
+        }>,
+    ): Promise<ApiHandlerResponse<UserWithProfileRelation>> {
+        return {
+            status: HttpCode.OK,
+            payload: await this.userService.updateUserProfileAndEmail(
+                options.params.id,
+                options.body,
```
Comment (Icerock10) [inline backend/src/bundles/users/user.controller.ts:176]: Put `try catch ` block here, and consider moving the service call into a variable

**Review by Icerock10 — CHANGES_REQUESTED** [review]

File: backend/src/bundles/users/user.service.ts (hunk 3 above)
Comment (Valentyn13) [reply backend/src/bundles/users/user.service.ts]: done

File: frontend/src/bundles/users/pages/profile.tsx (hunk 4 above)
Comment (Valentyn13) [reply frontend/src/bundles/users/pages/profile.tsx]: done

File: backend/src/bundles/users/user.controller.ts (hunk 5 above)
Comment (Valentyn13) [reply backend/src/bundles/users/user.controller.ts:176]: done

**Review by vlad-blip — APPROVED** [review]

**Comment by Valentyn13** [comment]
[Save / update data on profile page](https://trello.com/c/LsZwJoxX/129-save-update-data-on-profile-page)

**Pull request merged**